plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -570,18 +571,26 @@ function EpubDownloadBackend:createEpub(epub_path, html, url, include_images, me
     collectgarbage()
     collectgarbage()
 
+    local before_images_time = time.now()
+    local time_prev = before_images_time
+
     -- ----------------------------------------------------------------
     -- OEBPS/images/*
     if include_images then
         local nb_images = #images
         for inum, img in ipairs(images) do
-            -- Process can be interrupted at this point between each image download
+            -- Process can be interrupted every second between image downloads
             -- by tapping while the InfoMessage is displayed
             -- We use the fast_refresh option from image #2 for a quicker download
-            local go_on = UI:info((message and message ~= "" and message .. "\n\n" or "") .. T(_("Retrieving image %1 / %2 …"), inum, nb_images), inum >= 2)
-            if not go_on then
-                cancelled = true
-                break
+            local go_on
+            local now = time.now()
+            if time.to_ms(now - time_prev) > 1000 then
```

Comment:
There's a `time.since` wrapprt for this sort of stuff ;). 
```suggestion
            if time.since(time_prev) > 1.0 then
```

Reply:
To be clear, you'd prefer this? Because in principle the suggestion doesn't seem to make much sense to me.

```diff
diff --git a/plugins/newsdownloader.koplugin/epubdownloadbackend.lua b/plugins/newsdownloader.koplugin/epubdownloadbackend.lua
index 97cd84157..a833ed8bb 100644
--- a/plugins/newsdownloader.koplugin/epubdownloadbackend.lua
+++ b/plugins/newsdownloader.koplugin/epubdownloadbackend.lua
@@ -583,14 +583,13 @@ function EpubDownloadBackend:createEpub(epub_path, html, url, include_images, me
             -- by tapping while the InfoMessage is displayed
             -- We use the fast_refresh option from image #2 for a quicker download
             local go_on
-            local now = time.now()
-            if time.to_ms(now - time_prev) > 1000 then
+            if time.since(time_prev) > 1.0 then
                 go_on = UI:info((message and message ~= "" and message .. "\n\n" or "") .. T(_("Retrieving image %1 / %2 …"), inum, nb_images), inum >= 2)
                 if not go_on then
                     cancelled = true
                     break
                 end
-                time_prev = now
+                time_prev = time.now()
             end
             local src = img.src
             if use_img_2x and img.src2x then
```

Reply:
Also note that it'd have the sneaky side effect that then it'd be `> 1.1` in reality.

Reply:
Yep!